kamon-core/src/main/scala/kamon/ModuleSupervisor.scala

```diff
@@ -0,0 +1,49 @@
+package kamon
+
+import _root_.akka.actor
+import _root_.akka.actor._
+import kamon.ModuleSupervisor.CreateModule
+
+import scala.concurrent.{ Future, Promise }
+import scala.util.Success
+
+object ModuleSupervisor extends ExtensionId[ModuleSupervisorExtension] with ExtensionIdProvider {
+
+  def lookup(): ExtensionId[_ <: actor.Extension] = ModuleSupervisor
+  def createExtension(system: ExtendedActorSystem): ModuleSupervisorExtension = new ModuleSupervisorExtensionImpl(system)
+
+  case class CreateModule(name: String, props: Props, childPromise: Promise[ActorRef])
+}
+
+trait ModuleSupervisorExtension extends actor.Extension {
+  def createModule(name: String, props: Props): Future[ActorRef]
+}
+
+class ModuleSupervisorExtensionImpl(system: ExtendedActorSystem) extends ModuleSupervisorExtension {
+  import system.dispatcher
+  private val supervisor = system.actorOf(Props[ModuleSupervisor], "kamon")
+
+  def createModule(name: String, props: Props): Future[ActorRef] = Future {} flatMap { _: Unit ⇒
```

Comment:
I need to review this part again. There is a problem with the initialization order when using the Akka instrumentation, because the Dispatchers instrumentation might want to lookup the metrics extension before the ActorSystem is fully constructed and, the metrics extension wants the ModuleSupervisor to create the subscriptions actor, which can't be created because the actor system isn't fully constructed yet, causing a sad deadlock :(... 

I guess that we will need to separate the creation of the Metrics extension from the creation of the subscriptions dispatcher actor, so that the metrics extension is "akka free" and we don't have issues creating it at any point.